### PR TITLE
CDK3 Installer implementation

### PR DESCRIPTION
**Pull request by dgolovin**
Fix includes:
1. Vagrant removal;
2. New CDK 3 resources: minishift and ISO image
3. Updated unit tests

### CODE REVIEW

**Comment by jrichter1** [comment]
I'll add a few remarks on the code
- you probably forgot to update cdk in mac requirements, hence the build failure there
- some mentions of vagrant still present, namely
https://github.com/redhat-developer-tooling/developer-platform-install/blob/473ff8d1b4052bb6b5dac8b8583cebf59182394c/browser/model/helpers/installer.js#L41
https://github.com/redhat-developer-tooling/developer-platform-install/blob/473ff8d1b4052bb6b5dac8b8583cebf59182394c/browser/model/virtualbox.js#L223
and the uninstaller (no big deal if we keep it that way)
and some of my tests, which I should fix afterwards 
- it's 'Hyper-V' as in hype :)

**Comment by dgolovin** [comment]
Thanks for review @jrichter1! I fixed requirements errors you mentioned and hyper typos :)

**Comment by dgolovin** [comment]
Vanished latest traces of vagrant.

File: browser/pages/confirm/controller.js [hunk 1]
```diff
@@ -35,14 +35,10 @@ class ConfirmController {
...
-          $scope.checkboxModel.vagrant.selectedOption = 'install';
-        }
-        if($scope.checkboxModel.virtualbox.selectedOption == 'detected'
-          && !$scope.checkboxModel.virtualbox.hasOption('detected')) {
-          $scope.checkboxModel.virtualbox.selectedOption = 'install';
-        }
+        // if($scope.checkboxModel.virtualbox.selectedOption == 'detected'
+        //   && !$scope.checkboxModel.virtualbox.hasOption('detected')) {
+        //   $scope.checkboxModel.virtualbox.selectedOption = 'install';
+        // }
```
Comment (jrichter1) [inline browser/pages/confirm/controller.js]: Any particular reason for commenting this for virtualbox?

Comment (jrichter1) [inline browser/pages/confirm/controller.js]: Do we no longer require vbox to install cdk btw? (provided hyper-v is not yet supported in the installer)

Comment (dgolovin) [inline browser/pages/confirm/controller.js]: CDK 2 never was dependent on virtualbox at installation time, all vagrant operations like registering VM images and installing vagrant plug-ins were working fine without virtualbox present. But to run CDK 2 one would need virtualbox or hyper-v on windows.

Here I just removed automatic selection of virtualbox if CDK 3 is selected for installation. That would let to install CDK 3 and manually decide (until there is no detection) what one is going to use virtualbox or Hyper-V as virtualization

Comment (jrichter1) [inline browser/pages/confirm/controller.js]: Thanks for the clarification. However, it creates a weird situation on mac, see the screenshot:
screen shot 2017-01-10 at 16 10 52
It now allows to deselect virtualbox but doesn't let you proceed if you do.

Comment (dgolovin) [inline browser/pages/confirm/controller.js]: you right, I'll better keep it it was for CDK 2 until Hyper-V detection is implemented

Comment (dgolovin) [inline browser/pages/confirm/controller.js]: returned back virtualbox as a requirement for CDK 3 for this sprint

**Pull request merged**